spec/components/step_by_step_nav_spec.rb

```diff
@@ -257,7 +257,25 @@ def stepnav
   it "adds ga4 attributes" do
     render_component(steps: stepnav, ga4_tracking: true)
 
+    expected = {
+      "event_name": "navigation",
+      "type": "step by step",
+      "index": { "index_section": "1", "index_link": "1" },
+      "index_total": "2",
+      "section": "Step 1",
+    }
+
     assert_select ".gem-c-step-nav[data-ga4-expandable]"
     assert_select ".gem-c-step-nav[data-module='gemstepnav ga4-event-tracker']"
+    assert_select "#{step1} .gem-c-step-nav__link[href='/link1'][data-ga4-link='#{expected.to_json}']"
+
+    expected = {
+      "event_name": "navigation",
+      "type": "step by step",
+      "index": { "index_section": "3", "index_link": "1" },
```

Comment:
This is `index_section: 3` even though it's Step 2 because I think Step 1 has two sections, so Step 2 is section 3.

Reply:
> I believe this is `index_section: 3` even though it's Step 2 because in the test data Step 1 has two sections, so Step 2 is section 3.

Thanks for the explanation. It's slightly laborious, but could you make this clearer in the tests? e.g. have one test that checks the behaviour with simple data (two steps) and one with this slightly more complex scenario of one step containing two?